### PR TITLE
Add Path workaround for windows

### PR DESCRIPTION
**Pull request by bolognam**
See issue #200 for more details, specifically:
 - https://github.com/JuliaInterop/MATLAB.jl/issues/200#issuecomment-2150516087
 - https://github.com/JuliaInterop/MATLAB.jl/issues/200#issuecomment-2150775426

### CODE REVIEW

**Review by topolarity — APPROVED** [review]
Might be worth updating the README, if other users can confirm that this gets the newer versions of MATLAB working on Windows for them

It's likely that the macOS and Linux issues are similar, but they'll need separate fixes since each platform has a different search behavior for libraries.

**Comment by musm** [comment]
I wonder if there is a less invasive approach as a bulk modification of the environment variable, as I think has the potential to corrupt other libraries.

File: src/MATLAB.jl [hunk 1]
```diff
@@ -80,7 +80,10 @@ function __init__()
     if libmx_size > 0 # non-zero size library path
 
     # load libraries
-
+    # workaround for https://github.com/JuliaInterop/MATLAB.jl/issues/200
+    if Sys.iswindows()
+        ENV["PATH"] = string(matlab_libpath, ";", ENV["PATH"])
+    end
```
Comment (asinghvi17) [inline src/MATLAB.jl:86]: Can you run all of this under `withenv`?  Then you could wrap the code below, which uses Libdl, in a function that is called in `withenv`.

The code would look like this:
```julia
function load_matlab_libraries()
	# do the loading here
end

...

# in init
if Sys.iswindows()
withenv(load_matlab_libraries, "PATH" => string(matlab_libpath, ";", ENV["PATH"]))
else
load_matlab_libraries()
```

**Comment by topolarity** [comment]
I believe that `AddDllDirectory` would be the primary alternative (https://github.com/jheinen/GR.jl/issues/489 seemed to tread similar ground)

However, those directories won't be seen by a `LoadLibraryW` w/o flags [like the one in MATLAB](https://github.com/CppMicroServices/CppMicroServices/blob/899530e5b6929c90479b1b6040d2a530776fc8a4/framework/src/util/SharedLibrary.cpp#L114), which doesn't opt-in to the new search directories.

`SetDefaultDllDir` does change the LoadLibraryW behavior to use the added paths, but it's broken for e.g. SuiteSparse in Julia unless changes are made to Julia's library loading behavior: https://github.com/JuliaLang/julia/pull/47775

**Comment by musm** [comment]
@topolarity  I think we should explore that option in another PR. For now I will merge this since it at least allows us to use MATLAB. @bolognam  do you mind also  updating the readme?

**Pull request merged**